File: Heat/ModelBinders/PrincipalModelBinder.vb
Imports System.Security.Principal

Public Class PrincipalModelBinder
    Implements IModelBinder

    Public Function BindModel(controllerContext As ControllerContext, bindingContext As ModelBindingContext) As Object Implements IModelBinder.BindModel
        If IsNothing(controllerContext) Then
            Throw New ArgumentNullException("ControllerContext!")
        End If
        If IsNothing(bindingContext) Then
            Throw New ArgumentNullException("BindingContext!")
        End If

        Dim p As IPrincipal = controllerContext.HttpContext.User
        Return p

    End Function
End Class

File: Heat.ConvertedToC#/ModelBinders/PrincipalModelBinder.cs
using System;
using System.Web.Mvc;
using System.Security.Principal;
namespace Heat
{

    public class PrincipalModelBinder : IModelBinder
	{

		public object BindModel(ControllerContext controllerContext, ModelBindingContext bindingContext)
		{
			if ((controllerContext == null)) {
				throw new ArgumentNullException("ControllerContext!");
			}
			if ((bindingContext == null)) {
				throw new ArgumentNullException("BindingContext!");
			}

			IPrincipal p = controllerContext.HttpContext.User;
			return p;

		}
	}
}
